### ISSUE TITLE
[Bug] Protocol typing does not work for functions with decorators

### ISSUE THREAD

**Issue by PeterLaudel** [python/mypy#8698]
* Are you reporting a bug, or opening a feature request?
Bug
* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.
```python
from typing_extensions import Protocol
from functools import wraps

class SomeProtocol(Protocol):
    def some_func(self) -> bool:
        ...

class Something:
    def some_func(self) -> None:
        return

def my_decorator(func):
    @wraps(func)
    def wrapper():
        print("Before")
        func()
        print("After")

    return wrapper

@my_decorator
def extra_func(p: SomeProtocol) -> bool:
    return p.some_func()

extra_func(Something())
```
* What is the actual behavior/output?
```
Success: no issues found in 1 source files
```
* What is the behavior/output you expect?
```
src/test.py:29: error: Argument 1 to "extra_func" has incompatible type "Something"; expected "SomeProtocol"
src/test.py:29: note: Following member(s) of "Something" have conflicts:
src/test.py:29: note:     Expected:
src/test.py:29: note:         def some_func(self) -> bool
src/test.py:29: note:     Got:
src/test.py:29: note:         def some_func(self) -> None
```
* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?
Python: 3.7.7
mypy: 0.770
* What are the mypy flags you are using? (For example --strict-optional)
None
* If mypy crashed with a traceback, please paste
  the full traceback below.
Does not crash.

**Comment by JelleZijlstra** [comment]
I don't think this has anything to do with Protocols specifically; the problem is that your decorator is untyped. See https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators for an example declaring a decorator that does not change the function signature.

**Comment by PeterLaudel** [comment]
You are right. Iam really sorry I have overseen this, thanks for the hint and thank you for your time.
